calendar/store/user_store_test.go

```diff
@@ -0,0 +1,981 @@
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.setup(mockAPI)
+
+			user, err := store.LoadUser("mockMMUserID")
+
+			tt.assertions(t, user, err)
+
+			mockAPI.AssertExpectations(t)
+		})
+	}
+}
+
+func TestLoadMattermostUserID(t *testing.T) {
+	mockAPI, store, _, _, _ := GetMockSetup(t)
+
+	tests := []struct {
+		name       string
+		setup      func(*testutil.MockPluginAPI)
+		assertions func(*testing.T, string, error)
+	}{
+		{
+			name: "Error loading Mattermost User ID",
+			setup: func(mockAPI *testutil.MockPluginAPI) {
+				mockAPI.On("KVGet", mock.AnythingOfType("string")).Return(nil, &model.AppError{Message: "Load failed"})
+			},
+			assertions: func(t *testing.T, userID string, err error) {
+				require.Empty(t, userID)
+				require.EqualError(t, err, "failed plugin KVGet: Load failed")
+			},
+		},
+		{
+			name: "Success loading Mattermost User ID",
+			setup: func(mockAPI *testutil.MockPluginAPI) {
+				mockUserID := []byte("mockMattermostUserID")
+				mockAPI.ExpectedCalls = nil
+				mockAPI.On("KVGet", mock.AnythingOfType("string")).Return(mockUserID, nil)
+			},
+			assertions: func(t *testing.T, userID string, err error) {
+				require.NoError(t, err)
+				require.Equal(t, "mockMattermostUserID", userID)
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.setup(mockAPI)
+
+			userID, err := store.LoadMattermostUserID("mockRemoteUserID")
+
+			tt.assertions(t, userID, err)
+
+			mockAPI.AssertExpectations(t)
+		})
+	}
+}
+
+func TestLoadUserIndex(t *testing.T) {
+	mockAPI, store, _, _, _ := GetMockSetup(t)
+
+	tests := []struct {
+		name       string
+		setup      func(*testutil.MockPluginAPI)
+		assertions func(*testing.T, UserIndex, error)
+	}{
+		{
+			name: "Error loading UserIndex",
+			setup: func(mockAPI *testutil.MockPluginAPI) {
+				mockAPI.On("KVGet", mock.AnythingOfType("string")).Return(nil, &model.AppError{Message: "Load failed"})
+			},
+			assertions: func(t *testing.T, userIndex UserIndex, err error) {
+				require.Nil(t, userIndex)
+				require.EqualError(t, err, "failed plugin KVGet: Load failed")
+			},
+		},
+		{
+			name: "Success loading UserIndex",
+			setup: func(mockAPI *testutil.MockPluginAPI) {
+				mockAPI.ExpectedCalls = nil
+				mockUserIndexJSON := `[{"mm_username": "mockUser"}]`
+				mockAPI.On("KVGet", mock.AnythingOfType("string")).Return([]byte(mockUserIndexJSON), nil)
+			},
+			assertions: func(t *testing.T, userIndex UserIndex, err error) {
+				require.NoError(t, err)
+				require.Len(t, userIndex, 1)
+				require.Equal(t, "mockUser", userIndex[0].MattermostUsername)
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.setup(mockAPI)
+
+			userIndex, err := store.LoadUserIndex()
+
+			tt.assertions(t, userIndex, err)
+
+			mockAPI.AssertExpectations(t)
+		})
+	}
+}
+
+func TestLoadUserFromIndex(t *testing.T) {
+	mockAPI, store, _, _, _ := GetMockSetup(t)
```

Comment:
Defile it inside the loop. It can cause problems in future if someone else is writing the testcases. He might not realise he has to make expected calls nil